### ISSUE TITLE
Failed to parse segments error

### ISSUE THREAD

**Issue by xanthos84** [Sorrow446/Beatport-Downloader#1]
hi, i have tried to run the tool - but getting stuck at 502 bad gateway.
```
Signed in successfully - Beatport LINK

Album 1 of 1:
Adelphi Music Factory - Save Me (Club Mix)
Downloading track 1 of 1: Save Me (Club Mix) - AAC 256
Failed to parse segments.
502 Bad Gateway

```

Can you help whats wrong?
I also tried to run the go source code (but I have no go-knowledge) and so I end up with error:

root@ubuntu:/opt/Beatport-Downloader# go run main.go LINK_TO_BEATPORT
# command-line-arguments
./main.go:48:10: undefined: Transport

**Comment by Sorrow446** [comment]
Beatport patched it. I wanna fix it, but their Android app needs something like Android 11, and my emulator is only something like 9.

**Comment by xanthos84** [comment]
Oh I see.
And you need the android app to reverse-engineer the web calls?

**Comment by Sorrow446** [comment]
Yes, unpinning and sniffing is no problem. Just the version problem.

**Comment by Sorrow446** [comment]
Am able to sniff now, but the app says invalid credentials. Tried on an unrooted device, too.

**Comment by atarbig** [comment]
i have error to (windows x64)
Signed in successfully - Beatport LINK PRO

Album 1 of 1:
Magnus, Fancy INC - Heaven
Downloading track 1 of 1: Heaven (Extended Mix) - AAC 256
Failed to parse segments.
502 Bad Gateway

you can fix?

**Comment by mw-nfi** [comment]
Is there any update right now?  Can only be tested with Android?

**Comment by dpkiss** [comment]
Let me know if its possible to use this script on windows 10. and if Yes please somebody can help me?!

**Comment by atarbig** [comment]
For windows 10 still error 
Failed to parse segments.
502 Bad Gateway

**Comment by dpkiss** [comment]
> For windows 10 still error Failed to parse segments. 502 Bad Gateway

Thank you for your quick reply. So if want to run must to do this in ubuntu. Let me know if works for you în linux ubuntu to know if install this os in WM or not.

Thanks

**Comment by mw-nfi** [comment]
It is not due to windows linux or any other system. Beatport has changed the streaming quality in the web interface. You would have to read the 3rd party software TCP packets. I have tried with Rekodbox, also find the frames but can not decrypt the requests.

**Comment by fcollienne-afelio** [comment]
Same on macOS -> 502 Bad Gateway

**Comment by pee2ach** [comment]
would love to see an update :)

**Comment by Sorrow446** [comment]
Can fix it when I have an account to use.

**Comment by mw-nfi** [comment]
Even if you have the Pro plan, the web player only plays 128kbit/s.

So you would have to sniff Rekordbox or one of the other 3rd apps that support Link.

**Comment by Sorrow446** [comment]
Official mobile app also supports 256.

**Comment by mw-nfi** [comment]
Ah cool. Can you update it ? I've tested with rekordbox and serato. But no luck.

**Comment by mw-nfi** [comment]
Do you have an account? If you have Paypal, I can give you the link subscription for free for one month.

**Comment by Sorrow446** [comment]
I don't have PayPal, sorry. Would appreciate an already subscribed account.

**Comment by mw-nfi** [comment]
Are you using fiddler (Classing/Everywhere) for capture?

I could give you the captured logs? I would just need to know how to set the settings correctly.

**Comment by zoonoon** [comment]
i wanna share my account. how can i send you? not here.

**Comment by modernmediagrp** [comment]
need a pro plan? I'll hook you up.

**Comment by DeletedByAccident** [comment]
> When someone lends an account.

Hi,
I can lend you my account. I have the pro subscription which also comes with the 1000 track offline playback capability.

Just let me know how would you like me to share you the credentials :)

**Comment by lalauz** [comment]
anyone know how download in 256 kbps?

**Comment by modernmediagrp** [comment]
Gotta some how use the dj.beatport.com api, trying to first auth then switch but im no coder

**Comment by lalauz** [comment]
> 

Were you able to download any files? I managed to download it through the browser, but it's only 128 kbps files. I believe that in the android app or using programs like virtual dj or record box it is in 256 kbps aac

**Comment by modernmediagrp** [comment]
DM me. Yes I got a solution

**Comment by MattNguyen** [comment]
> DM me. Yes I got a solution

@modernmediagrp DM through what?

**Comment by DeletedByAccident** [comment]
> DM me. Yes I got a solution



> > DM me. Yes I got a solution
> 
> @modernmediagrp DM through what?

Could we form a telegram group or something?
We can exchange info through temp mail emails first as I don't like the idea of posting my IM handles publicly.

Would be awesome to get this to work again!

**Comment by DeletedByAccident** [comment]
> DM me. Yes I got a solution

Hi! Really interested in the fix if you can assist :) Github doesn't support DMs :/

**Comment by modernmediagrp** [comment]
check yarrm80s/orpheusdl
